### ISSUE TITLE
HEAD002 flags headings that begin with an inline macro

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#63]
## Problem

HEAD002 reports "Heading should start with uppercase letter" for headings that begin with an inline macro. The rule checks the first character of the first word, which here is the macro name (`image`), not visible text.

The arc42 template uses exactly this title pattern (`= image:arc42-logo.png[arc42] Template`), so every arc42-based document gets this warning. Seen in a downstream docs project.

## Minimal reproduction

At commit `911440ac35d5849349389cce7ba04311d75dca49`, with an existing `logo.png` next to `file.adoc`:

```asciidoc
= image:logo.png[Company logo] Project Title

== Section

text
```

```
asciidoc-linter --format plain file.adoc
```

## Expected vs. actual

- Expected: no HEAD002 finding, because the first visible word ("Project") is capitalised.
- Actual: `Heading should start with uppercase letter`.

## Proposal

Before checking capitalisation, skip leading inline macros (`image:`, `link:`, `xref:`, `kbd:`, `icon:` …) and attribute references (`{name}`), then check the first visible word. If nothing visible remains, skip the check.

**Comment by rdmueller** [comment]
## Analysis

Confirmed at 2ded7ad. `HeadingFormatRule.check_line()` (`asciidoc_linter/rules/heading_rules.py`) checks `words[0][0].islower()`. For `= image:logo.png[Company logo] Project Title`, the first "word" is `image:logo.png[Company`. Its first character is the macro name, not visible text.

**Fix plan (fix order: 3 of 9, small and independent)**
- Before checking case, strip leading inline macros (`name:target[...]`, including `image:`, `link:`, `xref:`, `kbd:`, `icon:`, `footnote:` …), attribute references (`{name}`) and anchors (`[[id]]`, `<<ref>>`) with a regex, one after another.
- Check the first remaining word. If nothing visible remains, skip the check.
- Only the *leading* macros are skipped. `= lower case title` is still reported.
- Tests: arc42 pattern, several leading macros, a title that is only a macro, attribute reference, lowercase text after a macro (must be reported).

---
_Generated by [Claude Code](https://claude.ai/code)_
